test/jdk/tools/jpackage/helpers/jdk/jpackage/test/JPackageCommand.java

```diff
@@ -816,22 +830,69 @@ public JPackageCommand assertImageCreated() {
         return this;
     }
 
-    JPackageCommand assertAppLayout() {
-        assertAppImageFile();
-        assertPackageFile();
-
-        TKit.assertDirectoryExists(appRuntimeDirectory());
-
-        if (!isRuntime()) {
-            TKit.assertExecutableFileExists(appLauncherPath());
-            TKit.assertFileExists(appLauncherCfgPath(null));
-
+    public static enum AppLayoutAssert {
+        APP_IMAGE_FILE(JPackageCommand::assertAppImageFile),
+        PACKAGE_FILE(JPackageCommand::assertPackageFile),
+        MAIN_LAUNCHER(cmd -> {
+            if (cmd.isRuntime()) {
+                TKit.assertPathExists(convertFromRuntime(cmd).appLauncherPath(), false);
+            } else {
+                TKit.assertExecutableFileExists(cmd.appLauncherPath());
+            }
+        }),
+        MAIN_LAUNCHER_CFG_FILE(cmd -> {
+            if (cmd.isRuntime()) {
+                TKit.assertPathExists(convertFromRuntime(cmd).appLauncherCfgPath(null), false);
+            } else {
+                TKit.assertFileExists(cmd.appLauncherCfgPath(null));
+            }
+        }),
+        RUNTIME_DIRECTORY(cmd -> {
+            TKit.assertDirectoryExists(cmd.appRuntimeDirectory());
             if (TKit.isOSX()) {
-                TKit.assertFileExists(appRuntimeDirectory().resolve(
-                        "Contents/MacOS/libjli.dylib"));
+                var libjliPath = cmd.appRuntimeDirectory().resolve("Contents/MacOS/libjli.dylib");
+                if (cmd.isRuntime()) {
+                    TKit.assertPathExists(libjliPath, false);
+                } else {
+                    TKit.assertFileExists(libjliPath);
+                }
             }
+        }),
+        MAC_BUNDLE_STRUCTURE(cmd -> {
+            if (TKit.isOSX()) {
+                MacHelper.verifyBundleStructure(cmd);
+            }
+        });
+        ;
```

Comment:
Remove ";"?

Reply:
I replaced `));` with `)),` and keep the terminating `;`